File: ApiProject/API-TFG/src/main/java/com/aitor/api_tfg/model/request/RegisterRequest.java
package com.aitor.api_tfg.model.request;

import com.aitor.api_tfg.model.db.Gender;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RegisterRequest {
    Integer id;
    String username;
    String email;
    String password;
    String firstname;
    String lastname;
    Double weight;
    Integer height;
    LocalDate birthdate;
    Gender gender;
}

File: AndroidProject/app/src/main/java/com/aitor/trackactividades/authentication/data/request/RegisterRequest.kt
package com.aitor.trackactividades.authentication.data.request

import com.aitor.trackactividades.authentication.presentation.model.Gender
import com.google.gson.annotations.SerializedName
import java.time.LocalDate

data class RegisterRequest (
    @SerializedName("username") val username: String,
    @SerializedName("email") val email: String,
    @SerializedName("password") val password: String,
    @SerializedName("firstname") val firstname: String,
    @SerializedName("lastname") val lastname: String,
    @SerializedName("weight") val weight: Double,
    @SerializedName("height") val height: Int,
    @SerializedName("birthdate") val birthdate: String,
    @SerializedName("gender") val gender: String
)